vosekast_control/TestSequence.py

```diff
@@ -31,134 +32,205 @@ def __init__(
+                    # get flow average
+                    flow_average = self.scale.flow_average()
+                    scale_actual = round(
+                        self.scale.scale_history[0] - scale_nulled, 5)
+
+                    # #write values to csv file
+                    # with open('sequence_values.csv', 'a', newline='') as file:
+                    #     writer = csv.writer(file, delimiter=',',
+                    #                     quotechar='|', quoting=csv.QUOTE_MINIMAL)
+                    #     writer.writerow([self.scale.scale_history[1], scale_actual, self.scale.flow_history[0], flow_average])
+
+                    # db
+                    dbconnect = sqlite3.connect('sequence_values.db')
+                    c = dbconnect.cursor()
+                    c.execute("""CREATE TABLE IF NOT EXISTS sequence_values (
+                        description text,
+                        timestamp real,
+                        scale_value real,
+                        flow_current real,
+                        flow_average_of_5 real
+                        )""")
 
-            await self.stop_sequence()
-            self.vosekast.constant_tank.state = self.vosekast.constant_tank.STOPPED
-            self.vosekast.state = "RUNNING"
+                    dbconnect.commit()
 
-    # todo fix drain
+                    try:
+                        c.execute("INSERT INTO sequence_values VALUES (:description, :timestamp, :scale_value, :flow_current, :flow_average)", {
+                            'description': "description",
+                            'timestamp': self.scale.scale_history[1],
+                            'scale_value': scale_actual,
+                            'flow_current': self.scale.flow_history[0],
+                            'flow_average': flow_average
+                        })
+
+                        dbconnect.commit()
+
+                    except Error as e:
+                        self.logger.warning(e)
+
+                    except:
+                        self.logger.warning("Error writing to db.")
+                        dbconnect.close()
+
+                    self.logger.debug(
+                        str(scale_actual) + " kg, flow rate (average) " + str(flow_average) + " L/s")
+                    await asyncio.sleep(1)
+
+                dbconnect.close()
+
+                # todo sqlite
+
+                # interrupt if measuring_tank full
+                if self.vosekast.measuring_tank.is_filled:
+                    self.vosekast.measuring_tank_switch.close()
+                    self.vosekast.measuring_tank.drain_tank()
+                    self.logger.debug(
+                        "Draining measuring tank, opening Measuring Tank bypass.")
+
+            except:
+                self.logger.warning("Write loop killed, stopping sequence.")
+                dbconnect.close()
+                await self.stop_sequence()
+        else:
```

Comment:
same here. no unnecessary code copy and paste.

Reply:
removed unnecessary code.